### PR TITLE
Fixed #29892 -- Fixed some selenium race conditions using wait over assert

### PR DESCRIPTION
**Pull request by codingjoe**
https://code.djangoproject.com/ticket/30974

### CODE REVIEW

**Comment by codingjoe** [comment]
buildbot, test on selenium.

**Comment by codingjoe** [comment]
buildbot, test on selenium.

**Comment by codingjoe** [comment]
buildbot, test on selenium.

**Comment by codingjoe** [comment]
> OK, I see where you're going with this. Not 100% convinced we can't reasonably assert, perhaps being more careful with our waits… (to discuss).

Yes, I did kind of fall into a rabbit hole here. Then again, we would consider race conditions bugs in any other part of the code base. Maybe the wisest approach is to fix the places where if fails often.
The once I fixed so far are the once that failed for me every often when running locally on and GitHub actions.

> Can you fix the `TypeError` failures?

Done

File: tests/admin_views/test_autocomplete_view.py [hunk 1]
```diff
@@ -185,20 +184,27 @@ def test_select(self):
         # Load next page of results by scrolling to the bottom of the list.
         for _ in range(len(results)):
             search.send_keys(Keys.ARROW_DOWN)
```
Comment (carltongibson) [inline tests/admin_views/test_autocomplete_view.py]: So, here... is there something we can wait on to say that the http request completed before we move on to the asserts below?

File: tests/admin_views/tests.py [hunk 2]
```diff
@@ -4637,9 +4637,9 @@ def test_first_field_focus(self):
         # First form field has a single widget
         self.admin_login(username='super', password='secret', login_url=reverse('admin:index'))
         self.selenium.get(self.live_server_url + reverse('admin:admin_views_picture_add'))
-        self.assertEqual(
```
Comment (carltongibson) [inline tests/admin_views/tests.py]: Similar here... Given #12051 can we wait for the load before making the assert?

**Comment by carltongibson** [comment]
I'm going to close this assuming that we'll go with #12152. (But shout there if you have more input!) Thanks for the effort on all these tickets!

**Pull request closed without merging**